ckanext/spectrum/actions.py

```diff
@@ -44,36 +45,60 @@ def package_create(next_action, context, data_dict):
 
     return next_action(context, data_dict)
 
+
 @toolkit.chained_action
 def user_create(next_action, context, data_dict):
-    """
-    Autogenerates a password and username (if password or username is not provided).
-    """
-
     if not data_dict.get('password'):
         data_dict['password'] = secrets.token_urlsafe(32)
 
-    if data_dict.get('name'):
-        return next_action(context, data_dict)
-
-    if not data_dict.get('email'):
-        raise toolkit.ValidationError(toolkit._("Must specify either a name or an email"))
+    if not data_dict.get('name'):
+        if not data_dict.get('email'):
+            raise toolkit.ValidationError(toolkit._("Must specify either a name or an email"))
+        else:
```

Comment:
I guess there's actually no need for the else.  `raise` is an exit point. Might make the code simpler to read.

Reply:
Yep, right, good catch :).